Run subsystem schema tests by default

kabir opened the pull request:
https://issues.jboss.org/browse/WFLY-5658

This and https://github.com/wildfly/wildfly/pull/8396 will need to be merged before https://github.com/wildfly/wildfly-core/pull/1250 is in a core release

kabir committed:
[WFCORE-1128] Make subsystem test testSchemaOfSubsystemTemplates() by

default

In cases where the test should not run, we override the
testSchemaOfSubsystemTemplates()
method to cut down on noise about ignored methods when running tests

It includes logic to test all the supplement varieties in the
subsystem template.

bstansberry commented on pom.xml:
```diff
@@ -216,7 +216,7 @@
         <version.org.wildfly.build-tools>1.1.0.Final</version.org.wildfly.build-tools>
         <version.org.wildfly.checkstyle-config>1.0.4.Final</version.org.wildfly.checkstyle-config>
-        <version.org.wildfly.core>2.0.1.Final</version.org.wildfly.core>
+        <version.org.wildfly.core>2.0.2.Final-SNAPSHOT</version.org.wildfly.core>
```
This won't work. :)

It doesn't seem to be needed though. I was able to revert this line and do a build, at least through the smoke tests.

kabir replied:
oops :-)

kabir committed:
[WFCORE-1128] Make subsystem test testSchemaOfSubsystemTemplates() by

default

In cases where the test should not run, we override the
testSchemaOfSubsystemTemplates()
method to cut down on noise about ignored methods when running tests

It includes logic to test all the supplement varieties in the
subsystem template.

bstansberry commented:
Looks good. Thanks.

I've put the 11.x label on this as I don't thinks it's needed for 10.

Last week there were some discussions around this general topic on the EAP side. If those lead to a desire to get this in the EAP branch for 7.0, then we can do a JBEAP and port this and the core PR.

bstansberry commented:
@kabir Is this still wanted? I forget where we came out re: running these tests.

bstansberry commented:
@kabir Is this still wanted? I forget where we came out re: running these tests.
